app/models/service_template.rb

```diff
@@ -437,11 +437,15 @@ def order(user_or_id, options = nil, request_options = nil, schedule_time = nil)
   def provision_workflow(user, dialog_options = nil, request_options = nil)
     dialog_options ||= {}
     request_options ||= {}
-    ra_options = { :target => self, :initiator => request_options[:initiator] }
-    ResourceActionWorkflow.new({}, user,
-                               provision_action, ra_options).tap do |wf|
+    ra_options = {
+      :target          => self,
+      :initiator       => request_options[:initiator],
+      :submit_workflow => request_options[:submit_workflow]
+    }
+
+    ResourceActionWorkflow.new(dialog_options, user, provision_action, ra_options).tap do |wf|
       wf.request_options = request_options
-      dialog_options.each { |key, value| wf.set_value(key, value) }
+      # dialog_options.each { |key, value| wf.set_value(key, value) }
```

Comment:
I know you just kinda took my diff and you're in the middle of testing, but you *may* need this line. If testing determines it works without it then we should just remove the commented line.